src/storage/mutate/AddEdgesProcessor.cpp

```diff
@@ -157,24 +156,27 @@ AddEdgesProcessor::addEdges(PartitionID partId,
                         return folly::none;
                     }
                     val = std::move(obsIdx).value();
-                }
-
-                if (!val.empty()) {
-                    auto reader = RowReaderWrapper::getEdgePropReader(env_->schemaMan_,
+                    if (!val.empty()) {
+                        oReader = RowReaderWrapper::getEdgePropReader(env_->schemaMan_,
```

Comment:
This is not like what you said in summary `execute get reader once`. You need to check if the edge has related index, if yes, then get reader.